### ISSUE TITLE
Verify API and import validation limits names to [\w] chars, and enforces data integrity

### ISSUE THREAD

**Issue by chouseknecht** [ansible/galaxy#429]
It should not be possible to create a namespace name, repository name, nor content name containing special characters.

- [x] Verify namespace.name only accepts [\w-], and that a validation error is surfaced by the API
- [x] ~~If the API encounters one or more '.' in the repository name, it should replace all '.' with '_', and store the new name as 'name', and the original name as 'original_name'.~~
- [x] Verify repository.name only accepts [\w-], and that a validation error is surfaced by the API
- [x] Verify that the import process throws a validation error, when content.name contains a character not in [\w-]
- [x] Verify there is a unique constraint on lowercase(namespace.name), and that a violation is surface by the API 
- [x] Verify there is a a unique constraint on lowercase(namespace.name, repository.name), and a violation is surfaced by the API
- [x] Verify the API enforces a unique constraint on lowercase(namespace.name, repository.name, content.name), and a violation is surfaced by the API

**Comment by chouseknecht** [comment]
Here's the query with regex used to identify invalid names currently in production (see #299): 

```
SELECT name, modified FROM main_content WHERE name !~ '^[\w-]+$';
```

**Comment by newswangerd** [comment]
last bullet fails. See #664

**Comment by chouseknecht** [comment]
Closed #664. If two roles or content items in a repo collide, I think for now, that's on the user, as it's difficult for us to detect. We may address this in the future with Ansible Lint or some other tool.

**Comment by chouseknecht** [comment]
To fix the last bullet item we need to make the following changes:

- [ ] in worker/tasks.py set the repository.name, to original_name lowercased and with [-.] converted to '_'.
- [ ] In worker/importers/base.py, change match expression on line #62 by removing '-'
- [ ] in main/models.py change the unique index on Content to include repository.name
